src/test/java/net/datafaker/service/FakeValuesServiceTest.java

```diff
@@ -241,6 +235,38 @@ void testLocalesChainGetterRu() {
         assertThat(chain).isEqualTo(processedChain);
     }
 
+    @Test
+    void testFakerContextSetLocale() {
+        final FakerContext fakerContext = new FakerContext(new Locale("en"), randomService);
+        fakerContext.setLocale(new Locale("uk"));
+        assertThat(fakerContext.getLocale()).isEqualTo(new Locale("uk"));
+    }
+
+    @Test
+    void testFakerContextSetRandomService() {
+        final FakerContext fakerContext = new FakerContext(Locale.US, randomService);
+        fakerContext.setRandomService(new RandomService(new Random(1)));
+        assertThat(fakerContext.getRandomService()).isEqualTo(new RandomService(new Random(1)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("fakerContexts")
+    void checkFakerContextEquality(FakerContext fc1, FakerContext fc2, boolean equals) {
+        if (equals) {
+            assertThat(fc1).isEqualTo(fc2);
+        } else {
+            assertThat(fc1).isNotEqualTo(fc2);
+        }
+    }
+
+    static Stream<Arguments> fakerContexts() {
+        return Stream.of(
+            Arguments.of(new FakerContext(new Locale("en"), new RandomService(new Random(1))), new FakerContext(new Locale("uk"), new RandomService(new Random(1))), false),
+            Arguments.of(new FakerContext(new Locale("en"), new RandomService(new Random(1))), null, false),
+            Arguments.of(new FakerContext(Locale.US, new RandomService(new Random(1))), new FakerContext(Locale.US, new RandomService(new Random(1))), true)
```

Comment:
I don't know why, but this one fails. Says that it's not equal

Reply:
randomservice and random do not have equals method

Reply:
Oh. In that case, maybe there's another way of implementing such test?

Reply:
compare by references instead of value

Reply:
Take a look please if this is good approach